test/testdata/resolver/optional_cyclic.rb

```diff
@@ -13,7 +13,7 @@ def bar(x: y, y: x) # error: Method `y` does not exist on `Test`
   end
 
   sig {params(x: String, y: Integer).returns(String)}
-  def qux(x: '', y: x) # error: Argument does not have asserted type `Integer`
+  def qux(x: '', y: x) # error: Returning value that does not conform to method result type
```

Comment:
Same comment as above, this is a confusing error message for this point in the program.